base-v3/src/main/java/com/adyen/checkout/base/model/payments/response/Threeds2Action.java

```diff
@@ -79,4 +84,48 @@ public String getToken() {
     public void setToken(@Nullable String token) {
         this.token = token;
     }
+
+    @Nullable
+    public String getSubtype() {
+        return subtype;
+    }
+
+    public void setSubtype(@Nullable String subtype) {
+        this.subtype = subtype;
+    }
+
+    @Nullable
+    public SubType getSubtypeEnum() {
+        return parseStringToSubType(subtype);
+    }
+
+    @Nullable
+    private SubType parseStringToSubType(@Nullable String type) {
+        if (type == null) {
+            return null;
+        }
+        switch (type) {
+            case SubTypes.FINGERPRINT:
+                return SubType.FINGERPRINT;
+            case SubTypes.CHALLENGE:
+                return SubType.CHALLENGE;
+            default:
+                return null;
+        }
+    }
+
+    private static final class SubTypes {
```

Comment:
What do you think about passing the token name as an argument in the `enum` itself instead of having this separate class for it?

Reply:
Hmmm we would still need to have theses 2 constants (`FINGERPRINT` and `CHALLENGE `) somewhere though right?

Reply:
Yeah but they can be on the top level class, I just think this `SubTypes` class is a bit unnecessary.